report/report.go

```diff
@@ -111,3 +110,72 @@ func (r Report) LocalNetworks() []*net.IPNet {
 	}
 	return ipNets
 }
+
+// OriginTable produces a table (to be consumed directly by the UI) based on
+// an origin ID, which is (optimistically) a node ID in one of our topologies.
+func (r Report) OriginTable(originID string) (Table, bool) {
```

Comment:
I'm not fussed about free function or not, more that this function is only used from detailed_node.go (I think), so I guess it belongs there.  But I just realised they are in the same package so the receiver doesn't matter.

OTOH you could use the same argument about MakeDetailedNode; having both of these in the public API for a report seems weird.